### PR TITLE
Add Schwab bank support to schwab_csv importer

### PR DESCRIPTION
**Pull request by bayesianmind**
I have quite a lot of activity / business with Schwab including a checking account so I brought in support for everything I could find in my own Schwab CSV files. In the spirit of #88 .

Other changes in the schwab_csv source: (apologies for not splitting these out, if that's a bother I can do the extra book keeping to get them into independent pull reqs).
- Added support for 7 new BrokerageActions.
- Added support for fractional shares, which Schwab added last year. You can get these in your account via dividend re-investment or the fractional share purchase feature. The source previously assume int for share quantities.
- Make schwab_csv post cleared status for Schwab accounts. This lets uncleared transactions properly appear in the import webui.
- Make the currency configurable just in case instead of hardcoding USD.
- Added a dedicated interest type that we now use consistently for Checking account interest and "Bank interest" paid on brokerage account cash balances.
- Tweaked some of the key regexps to work on the files I am getting from Schwab in addition to the testfiles.

Known caveats:
- The banking CSVs indicate they may contain pending transactions in a special way. I didn't have any in my file to add support for. Likely the importer will import them the same whether they are pending or not. This could result in some extraneous imports for pending auth charges that would normally drop off. If someone donates a CSV with the format it would be easy to add support for it.
- Does not support corporate equity awards accounts - for Google's at least it is quite different from the Brokerage format and I had written my own importer before schwab_csv was added. If there's interest I can open source the Google equity awards source.

### CODE REVIEW

**Comment by Zburatorul** [comment]
Did you forget to include a commit with more code changes? I don't see the 7 new brokerage actions.

**Comment by bayesianmind** [comment]
See line 134 and on. I didn't have to add new classes for them, they're similar enough to existing types.

File: beancount_import/source/schwab_csv.py
Comment (carljm) [reply beancount_import/source/schwab_csv.py:1059]: If we do leave this in, I'd rather handle it the same as `source_desc` (get the value with an empty default and return `None` if it is empty, rather than checking if the key is there and then later getting the value for that key.)

File: beancount_import/source/schwab_csv.py
Comment (carljm) [inline beancount_import/source/schwab_csv.py:1133]: It feels a bit risky to me to silently skip any row with something in the date column that doesn't look to us like a date. Never know when Schwab might do something funky in that column (like in the brokerage side sometimes it looks like `DATE as of DATE`, which is handled by `_convert_date`), and when we see something unexpected, we should fail loudly and clearly, not silently skip transactions. Can we enumerate here the contents of the date column for the rows you want to skip, instead of skipping any row where it doesn't look like your regex?

(Edit: this intuition is confirmed by the fact that it looks like this is in fact currently causing us to silently skip a transaction in the misformatted test CSV file.)

File: testdata/source/schwab_csv/test_basic/transactions/EAC_Checking_Transactions_20201223-160009.CSV
Comment (carljm) [inline testdata/source/schwab_csv/test_basic/transactions/EAC_Checking_Transactions_20201223-160009.CSV:4]: Something looks mis-formatted here. Is that coming directly from Schwab, or was accidentally introduced in this test file?
 
If the latter, let's re-verify the test file looks just like the CSV files you get from upstream.

If the former, that's unfortunate, but the importer needs adjusting to account for it so we don't skip the first transaction in every CSV. This row is currently silently ignored.

File: beancount_import/source/schwab_csv.py
Comment (carljm) [inline beancount_import/source/schwab_csv.py:1129]: It looks like all the source line numbers in the metadata for the generated transactions in the test are off by one (too high). Maybe this just needs to be `transaction_start_line = 1`?

File: beancount_import/source/schwab_csv.py
Comment (carljm) [inline beancount_import/source/schwab_csv.py:120]: This is much better, thanks for pulling out this constant.

**Review by carljm — CHANGES_REQUESTED** [review]
This looks great overall! Thanks so much for the additions and improvements. Made a few comments but overall I think this is pretty close to ready.

One thing is I noticed a few bad type annotations; haven't tested but I would think these should cause a mypy failure. Can you verify that `tox` is green with your changes?

File: beancount_import/source/schwab_csv.py
Comment (carljm) [inline beancount_import/source/schwab_csv.py:36]: Oh, minor nit but can you also revise the prose sentence immediately preceding this example to read "... for recording dividends, interest, capital gains, fees, and taxes." (Addition of "interest," in the list.)

File: beancount_import/source/schwab_csv.py
Comment (carljm) [inline beancount_import/source/schwab_csv.py:63]: Could you also add a note here about the possibility of `ValueError: 'Foo' is not a valid BankingEntryType`? Unless you're confident that Schwab will never add any banking entry types you haven't already seen.

**Comment by carljm** [comment]
I should also clarify that @jbms is the author and maintainer of beancount-import and the ultimate authority on whether this gets merged; just offering my review as the initial contributor of the Schwab CSV importer.

**Comment by bayesianmind** [comment]
Thanks for the excellent review @carljm ! I will update this over the next few days as I have free cycles.

File: beancount_import/source/schwab_csv.py
Comment (zhou13) [inline beancount_import/source/schwab_csv.py:1320]: The quantity of reinvested portfolios can be decimal. I need to replace this line with `quantity = _convert_decimal(row["Quantity"])`.

**Review by bayesianmind — COMMENTED** [review]
Thank you both so much for your feedback. This PR is way better for it.
It's ready to go now. When combined with #96  tox is passing.

**Comment by bayesianmind** [comment]
@jbms This is ready to merge alongside #96 when you get a chance. Thanks!

**Pull request merged**
